### ISSUE TITLE
CascadeSelect: Cascade Submenu is not focussable with right and left arrow navigations

### ISSUE THREAD

**Issue by vincyrose1970** [primefaces/primereact#8345]
### Describe the bug

CascadeSelect: Cascade Submenu is not focussable with right and left arrow navigations

The issue is reproducable in the main website itself, so not providing a stackblitz project.

https://primereact.org/cascadeselect/#basic

### Reproducer

https://primereact.org/cascadeselect/#basic

### System Information

```Shell
System:
    OS: Windows 11 10.0.26100
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
    Memory: 2.17 GB / 15.41 GB
  Binaries:
    Node: 22.12.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.9.0 - C:\Program Files\nodejs\npm.CMD
    pnpm: 10.15.0 - C:\Users\AkshayAntony\AppData\Roaming\npm\pnpm.CMD
  Browsers:
    Chrome: 141.0.7390.76
    Edge: Chromium (140.0.3485.54)
    Firefox: 143.0 - C:\Program Files\Mozilla Firefox\firefox.exe
    Internet Explorer: 11.0.26100.1882
  npmPackages:
    react: 18.3.1 => 18.3.1
```

### Steps to reproduce the behavior

Go to https://primereact.org/cascadeselect/#basic
Click on a menu item with dropdown say 'Select a city'
Try to navigate using arrow keys. Use down arrow to navigate through items and use right arrow key to navigate to the submenu. 
The focus is not pointing to the items in sub menu and navigation is not working.

Since focus is not going to the second submenu, I'm not able to navigate to the third and fourth submenu's if they exist. 

The reverse option is also not working, ie, using left arrow keys to navigate back to the main menu.

https://github.com/user-attachments/assets/6e0484ca-6ba2-4698-bedd-1a4b1112bf87

### Expected behavior

The focus should point to the items in sub menu and user should be able to navigate using arrow keys.
